Feature/create form

petia-latinova opened the pull request.

imollov commented on batch-nft-reveal/app/components/create/Create.tsx:
```diff
@@ -0,0 +1,312 @@
+  const [vrfGasLane, setVrfGasLane] = useState("");
+  const [vrfCallbackGasLimit, setVrfCallbackGasLimit] = useState("");
+
+  const [tokenNameEntered, setTokenNameEntered] = useState(false);
+  const [symbolEntered, setSymbolEntered] = useState(false);
+  const [maxSupplayEntered, setMaxSupplyEntered] = useState(false);
+  const [mintCostEntered, setMintCostEntered] = useState(false);
+  const [revealBatchSizeEntered, setRevealBatchSizeEntered] = useState(false);
+  const [revealIntervalEntered, setRevealIntervalEntered] = useState(false);
+  const [vrfCoordinatorV2Entered, setVrfCoordinatorV2Entered] = useState(false);
+  const [vrfSubscriptionIdEntered, setVrfSubscriptionIdEntered] = useState(false);
+  const [vrfGasLaneEntered, setVrfGasLaneEntered] = useState(false);
+  const [vrfCallbackGasLimitEntered, setVrfCallbackGasLimitEntered] = useState(false);
```
All this state can be computed.

imollov commented:
@petia-latinova the form must look like in the mockup. Currently, there are 1) missing fields, 2) unnecessary fields and 3) missing elements (divider, title).

imollov commented on batch-nft-reveal/app/components/create/Create.tsx:
```diff
@@ -0,0 +1,312 @@
+
+  const [tokenName, setTokenName] = useState("");
+  const [symbol, setSymbol] = useState("");
+  const [maxSupplay, setMaxSupply] = useState("");
```
```suggestion
  const [maxSupply, setMaxSupply] = useState("");
```

imollov commented on batch-nft-reveal/app/components/create/Create.tsx:
```diff
@@ -0,0 +1,312 @@
+        {isInvalidName && <FormErrorMessage>Name is not valid.</FormErrorMessage>}
+      </FormControl>
+
+      <FormControl isInvalid={symbolEntered && isInvalidSymbol}>
```
```suggestion
      <FormControl mt="4" isInvalid={symbolEntered && isInvalidSymbol}>
```

imollov commented on batch-nft-reveal/app/components/create/Create.tsx:
```diff
@@ -0,0 +1,312 @@
+  const isInvalidName = !tokenName;
+  const isInvalidSymbol = !symbol;
+  const isInvalidMaxSupplay = isNotNumber(maxSupplay) || Number(maxSupplay) <= 0;
+  const isInvalidMintCost = isNotNumber(mintCost); // TODO now is in satoshi... maybe make it fractional number
```
There is no satoshi unit on Ethereum. The cost is in ETH, as you can see in the mockup.

petia-latinova commented:
> @petia-latinova tests are failing, might be because `yarn.lock` is changed. Is there any reason for that?

The tests were failing and the versel deploy too and I deleted them and generated again to see if that will fix the problem. I removed them now. You are right, there is no reason to be changed.

imollov commented:
@petia-latinova pls check this best practice and refactor: https://refactoring.com/catalog/removeDoubleNegative.html

imollov commented:
@petia-latinova same as load page, submit button is not enabled when all fields are valid.

imollov commented:
> @petia-latinova same as load page, submit button is not enabled when all fields are valid.

It's because there's no account connected. Maybe we should add a tooltip in that case?

imollov commented on batch-nft-reveal/app/components/create/Create.tsx:
```diff
@@ -0,0 +1,272 @@
+  return (
+    <form onSubmit={formik.handleSubmit}>
+      <Container maxW="mt">
+        <FormikProvider value={formik}>
```
What's the purpose of this provider?

petia-latinova replied:
Without `FormikProvider` the validations in formik will not be working.
`const formik = useFormik(...)` creates the state for formik with all it's functionality.
The provider connects the variable `formik` to the web code inside it.

There are other ways to use formik without `FormikProvider` but with the provider you can use other states and hooks which are not working with some of the other ways you can use formik.

imollov commented on batch-nft-reveal/app/components/create/Create.tsx:
```diff
@@ -0,0 +1,272 @@
+
+  return (
+    <form onSubmit={formik.handleSubmit}>
+      <Container maxW="mt">
```
This should not be part of the form component.

imollov commented on batch-nft-reveal/app/components/create/Create.tsx:
```diff
@@ -0,0 +1,272 @@
+                Cost(ETH) {formik.errors.mintCost}
+              </FormErrorMessage>
+            )}
+            <span></span>
```
?

imollov commented on batch-nft-reveal/app/components/create/Create.tsx:
```diff
@@ -0,0 +1,272 @@
+              </FormErrorMessage>
+            )}
+          </FormControl>
+          <Tooltip
```
How it is:
<img width="192" alt="Screenshot 2022-08-24 at 17 51 27" src="https://user-images.githubusercontent.com/482351/186450392-74212f94-e41e-43f3-ae93-b34efef623d2.png">

How it should be:
<img width="147" alt="Screenshot 2022-08-24 at 17 53 30" src="https://user-images.githubusercontent.com/482351/186450868-0a79cf86-e672-4d54-98b2-7055cc6b06c5.png">
